File: src/main/java/com/epicdragonworld/gameserver/managers/IdManager.java
package com.epicdragonworld.gameserver.managers;

/**
 * @author Pantelis Andrianakis
 */
public class IdManager
{
	private static volatile long _lastId = 0;
	
	public static synchronized long getNextId()
	{
		return _lastId++;
	}
}

File: EpicDragonWorldServer/Managers/IdManager.cs
using System.Threading;

/**
 * Author: Pantelis Andrianakis
 * Date: November 7th 2018
 */
public class IdManager
{
    private static long LAST_ID = 0;

    public static long GetNextId()
    {
        return Interlocked.Increment(ref LAST_ID);
    }
}
